server/routes/appointments/create/handlers/uploadPrisonerList.ts

```diff
@@ -0,0 +1,101 @@
+import { Request, Response } from 'express'
+import { Expose } from 'class-transformer'
+import { IsNotEmpty } from 'class-validator'
+import { parse } from 'csv-parse'
+import fs from 'fs'
+import PrisonService from '../../../../services/prisonService'
+import IsNotEmptyFile from '../../../../validators/isNotEmptyFile'
+import IsValidCsvFile from '../../../../validators/isValidCsvFile'
+
+export class PrisonerList {
+  @Expose()
+  @IsNotEmpty({ message: "Select a CSV file of prisoners' numbers" })
+  @IsNotEmptyFile({ message: 'The selected file is empty' })
+  @IsValidCsvFile({ message: 'The selected file must be a CSV' })
+  file: Express.Multer.File
+}
+
+export default class UploadPrisonerListRoutes {
+  constructor(private readonly prisonService: PrisonService) {}
+
+  GET = async (req: Request, res: Response): Promise<void> => {
+    res.render(`pages/appointments/create/upload-prisoner-list`)
+  }
+
+  POST = async (req: Request, res: Response): Promise<void> => {
+    const prisonerListCsvFile = req.file
+    const { user } = res.locals
+
+    const prisonerNumbers: string[] = []
+    const errors: string[] = []
+
+    fs.readFile(prisonerListCsvFile.path, (err, result) => {
```

Comment:
I'd prefer using the fs promise api here

Reply:
Switched to promise API